test/modify.t

```diff
@@ -0,0 +1,158 @@
+        self.t("stop {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(hours=1)))
+        self.t("start {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(minutes=49)))
+        self.t("stop")
+
+        code, out, err = self.t.runError("modify stop @2 {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(hours=4)))
+        self.assertIn("Cannot modify interval", err);
+
+    def test_modify_start_to_after_end(self):
+        """Advance the start to after the stop time."""
+        now_utc = datetime.now().utcnow()
+
+        self.t("start {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(hours=3)))
+        self.t("stop {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(hours=1)))
+        self.t("start {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(minutes=49)))
+        self.t("stop")
+
+        code, out, err = self.t.runError("modify start @2 {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(minutes=59)))
+        self.assertIn("Cannot modify interval", err);
+
+    def test_modify_start_within_interval(self):
+        """Increase start time within interval."""
+        now_utc = datetime.now().utcnow()
+
+        self.t("start {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(hours=3)))
+        self.t("stop {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(hours=1)))
+        self.t("start {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(minutes=49)))
+        self.t("stop")
+
+        code, out, err = self.t("modify start @2 {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(hours=2)))
+
+        code, out, err = self.t("summary")
+        line = out.splitlines()[-2].strip(" ").rpartition(":")[0] # Get the total hours / minutes
+        self.assertEqual("1:49", line)
+
+    def test_modify_move_stop_to_overlap_following_interval(self):
+        """Move stop time to overlap with following interval."""
+        now_utc = datetime.now().utcnow()
+
+        self.t("start {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(hours=3)))
+        self.t("stop {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(hours=1)))
+        self.t("start {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(minutes=49)))
+        self.t("stop")
+
+        code, out, err = self.t.runError("modify stop @2 {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(minutes=30)))
+        self.assertIn("You cannot overlap intervals", err)
+
+    def test_modify_move_start_to_overlap_preceeding_interval(self):
+        """Move start time to overlap with preceeding interval."""
+        now_utc = datetime.now().utcnow()
+
+        self.t("start {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(hours=3)))
+        self.t("stop {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(hours=1)))
+        self.t("start {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(minutes=49)))
+        self.t("stop")
+
+        code, out, err = self.t.runError("modify start @1 {:%Y-%m-%dT%H:%M:%S}Z".format(now_utc - timedelta(hours=2)))
+        self.assertIn("You cannot overlap intervals", err)
+
```

Comment:
Test `modify should not move start beyond end`?
Test `modify should not move end before start`?

Reply:
I believe these case are already covered?  I've updated the test description to be of this form in the updated pull request.